onboarding/main.go

```diff
@@ -70,6 +70,8 @@ func main() {
 			},
 		}
 
+		clientset.CoreV1().Secrets(operatorNamespace).Delete(context.Background(), onboardingPrivateKeySecretName, metav1.DeleteOptions{})
```

Comment:
Why do we need this? what is the reason? and why outside the reconcile function?

Reply:
That does not make sense. if the public key is deleted then a reconcile will happen and both keys will be regenerated. I don't see the edge case here

Reply:
Okay, let me describe it more clearly.
- cluster is up and running and it has both public & private key
- Now if the admin deletes the public key by mistake then what should happen?
- It will reconcile but when we call clientset.CoreV1().Secrets(operatorNamespace).Create on private key it will return an error saying the secret already exists.

Reply:
I might be missing something here, but why do you want to call `clientset.CoreV1().Secrets(operatorNamespace).Create` on the secret? 

First, why are you working with clientset and not with a kubebuilder client? 
Second, in case the secret exists you should call `Update` and not `Create. The kube util library has a convenient `CreateOrUpdate` function that does exactly that!  (with the benefit of not actually calling the update if nothing changed)

Reply:
I have not used the Kubebuilder client because Leela suggested it's a small job bringing the heavyweight controller runtime is not good. Leela mentioned [here](https://github.com/red-hat-storage/ocs-operator/pull/2301#discussion_r1422415217). 
Yes, I agree instead of deleting the secret, I have to update if the secret already exists.

Reply:
Ok. But regardless of the client, you will want to replicate the same pattern `CreateOrUpdate` implements

Reply:
I have addressed the comments, please resolve them?